lib/compress/zstd_compress.c

```diff
@@ -279,7 +279,7 @@ static ZSTD_paramSwitch_e ZSTD_resolveEnableLdm(ZSTD_paramSwitch_e mode,
 
 /* Enables validation for external sequences in debug builds. */
 static int ZSTD_resolveExternalSequenceValidation(int mode) {
-#if defined(DEBUGLEVEL) && (DEBUGLEVEL>=2)
+#if defined(DEBUGLEVEL) && (DEBUGLEVEL>=1)
```

Comment:
The purpose of `DEBUGLEVEL==1` is to enable `assert()`.

Enabling `assert()` should never change the behavior of the program.
`assert()` are only meant to catch programming errors that should never happen.
They are not meant to "validate user input" for example.

Here we have a subtle cousin, when an assert-related constant (`DEBUGLEVEL`) is used to mean "validate user input".

I think it's not correct, and lead to other complexities, such as test conditions.

Probably something to discuss.

Reply:
I think keeping it at `>=2` is reasonable.

Reply:
@embg Another option would be to add an `assert(sequencesValid())` after the normal sequence validation check.

Basically add an `assert(!ZSTD_isError(ZSTD_validateSequences(...)));` here

https://github.com/facebook/zstd/blob/2f74507bbd0e8aca907a2842a311cc24ec815cf2/lib/compress/zstd_compress.c#L6276

Since it is UB to pass invalid sequences without validating them, we are within our right to assert that they are valid after we do the check.

Reply:
I'll add validateSequences to the fuzzer, though, so that when we expose the fuzzing interface for external matchfinders we'll be able to catch invalid sequence bugs.